creator-node/src/components/tracks/tracksComponentService.js

```diff
@@ -0,0 +1,187 @@
+const path = require('path')
+
+const { successResponse, errorResponseBadRequest, errorResponseServerError, errorResponseForbidden } = require('../../apiHelpers')
+const { logger: genericLogger } = require('../../logging')
+const { getSegmentsDuration } = require('../../segmentDuration')
+const TranscodingQueue = require('../../TranscodingQueue')
+const models = require('../../models')
+const DBManager = require('../../dbManager')
+const { removeTrackFolder, saveFileToIPFSFromFS } = require('../../fileManager')
+
+const SaveFileToIPFSConcurrencyLimit = 10
+
+/**
+ * Upload track segment files and make avail - will later be associated with Audius track
+ *
+ * Also the logic used in /track_content route. Params are extracted to keys that are necessary for the job so that
+ * we can pass this task into a worker queue.
+ *
+ * @param {Object} logContext the context of the request used to create a generic logger
+ * @param {Object} req more request specific context
+ * @returns a success or error server response
+ * @dev - Prune upload artifacts after successful and failed uploads. Make call without awaiting, and let async queue clean up.
+ */
+const handleTrackContentRoute = async ({ logContext }, req, ipfs, blacklistManager) => {
```

Comment:
maybe something we fast follow in v1 of this is ripping out req. having bull spin up a new worker won't work with req being passed in like this

Reply:
if you see how I called handleTrackContentRoute, I take out the fields necessary for this function to work https://github.com/AudiusProject/audius-protocol/pull/1452/files#diff-cea0356d19f998f416dcaca2141fb24156c529095b61daebb71f8e3316071f32R41

Reply:
the req object in this case is not the actual express req object, but a simplified version that is serializable

Reply:
Oh this is perfect, let's rename this to something like requestProps and add documentation?

Reply:
Basically I want to remove mentions of req here cause that's conflating this with the actual req obj